File: src/main/java/net/onelitefeather/titan/function/TitanFunction.java
package net.onelitefeather.titan.function;

import net.minestom.server.entity.Player;
import org.togglz.core.user.SimpleFeatureUser;

public interface TitanFunction {
    default SimpleFeatureUser toUser(Player player) {
        return new SimpleFeatureUser(player.getUsername());
    }
}

File: src/main/kotlin/net/onelitefeather/titan/function/TitanFunction.kt
package net.onelitefeather.titan.function

import net.minestom.server.entity.Player
import org.togglz.core.user.FeatureUser
import org.togglz.core.user.SimpleFeatureUser

interface TitanFunction {

    fun initialize()
    fun terminate()

    fun Player.toFeatureUser(): FeatureUser {
        return SimpleFeatureUser(this.username)
    }
}